contracts/base/BaseLiquidityManagement.sol

```diff
@@ -249,7 +316,8 @@ contract BaseLiquidityManagement is IBaseLiquidityManagement, SafeCallback {
         returns (BalanceDelta delta)
     {
         delta = _collect(owner, range, hookData);
-        zeroOut(delta, range.key.currency0, range.key.currency1, owner, claims);
+        _closeCallerDeltas(delta, range.poolKey.currency0, range.poolKey.currency1, owner, claims);
+        _closeAllDeltas(range.poolKey.currency0, range.poolKey.currency1);
```

Comment:
in decrease/collect scenarios (where we send positive deltas to the user), this ordering doesnt smell right to me..

there might be a case where we should collect tokens to posm first _then_ send them to the user

Reply:
why is that?

Reply:
I guess in my head the idea is that the `callerDelta` should always be the correct number that is owed to the caller or owed from the caller. 

and the _closeAllDeltas just uses the number open in core. so if callerDelta is up to date then the number queried in core should be up to date.

the alternative design is that we calculate exactly what THIS contract is owed or should owe, do that closing first, and then give the caller the leftover but that felt weirder